File: 09.Mass.Extinction.Discord.Bot/data/entities/Message.js
const { Model, DataTypes } = require("sequelize");

class Message extends Model {
    static sequelizeInit() {
        return {
            id: {
                field: "MessageId",
                type: DataTypes.INTEGER,
                primaryKey: true,
                autoIncrement: true
            },
            sender: {
                field: "Sender",
                type: DataTypes.TEXT
            },
            body: {
                field: "Body",
                type: DataTypes.TEXT
            },
            dateSent: {
                field: "DateSent",
                type: DataTypes.DATE
            },
            isAnonymous: {
                field: "IsAnonymous",
                type: DataTypes.BOOLEAN
            }
        }
    }
}

module.exports = Message;

File: 09.Mass.Extinction.Data/Entities/Message.cs
namespace Ninth.Mass.Extinction.Data.Entities;

public class Message
{
    public long Id { get; set; }
    public string Sender { get; set; }
    public string Body { get; set; }
    public DateTime DateSent { get; set; }
    public bool IsAnonymous { get; set; }
}
